src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs

```diff
@@ -52,6 +52,7 @@ public static IWebHostBuilder UseIISIntegration(this IWebHostBuilder hostBuilder
                 {
                     // Delay register the url so users don't accidently overwrite it.
                     hostBuilder.UseSetting(WebHostDefaults.ServerUrlsKey, address);
+                    hostBuilder.PreferHostingUrls(true);
```

Comment:
It gets rid of this problem:
https://github.com/aspnet/Security/blob/dev/samples/SocialSample/Program.cs#L19